### ISSUE TITLE
Scheduler perf incorrectly shows percentiles for fast metrics

### ISSUE THREAD

**Issue by macsko** [kubernetes/kubernetes#127384]
### What happened?

When all metric values are in the first bucket, percentiles are incorrectly computed:
```json
{
  "data": {
    "Average": 0.0009773531999999999,
    "Perc50": 0.05,
    "Perc90": 0.09000000000000001,
    "Perc95": 0.095,
    "Perc99": 0.099
  },
  "unit": "ms",
  "labels": {
    "Metric": "scheduler_framework_extension_point_duration_seconds",
  }
}
```

### What did you expect to happen?

Correctly computed percentiles. Perhaps bucketing should be changed for some metrics. 

### How can we reproduce it (as minimally and precisely as possible)?

Run scheduler_perf or see some [perf-dash](https://perf-dash.k8s.io/#/?jobname=scheduler-perf-benchmark&metriccategoryname=Scheduler&metricname=BenchmarkPerfResults&Metric=scheduler_framework_extension_point_duration_seconds&Name=SchedulingBasic%2F5000Nodes_10000Pods%2Fnamespace-2&extension_point=Permit&plugin=not%20applicable&result=not%20applicable&event=not%20applicable) charts.

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by macsko** [comment]
cc: @sanposhiho @pohly

**Comment by sanposhiho** [comment]
Is it allowed to change buckets when it comes to the stable metrics? (i.e., wouldn't the change be breaking change?)

**Comment by sanposhiho** [comment]
@kubernetes/sig-instrumentation-members Can anyone give us a suggestion for my above question?

**Comment by CatherineF-dev** [comment]
> When all metric values are in the first bucket, percentiles are incorrectly computed:

qq: what are your expected metric values? These Percentiles look okay.

**Comment by macsko** [comment]
> > When all metric values are in the first bucket, percentiles are incorrectly computed:
> 
> qq: what are your expected metric values? These Percentiles seem okay.

In this case percentiles are equal to `percentile * first bucket upper bound`, i.e. Perc99: `0.99 * 0.1 ms = 0.099` in this case. So, those percentiles are not even related to the gathered metric values right now.

**Comment by felipeagger** [comment]
I would like to contribute to this issue, but I don't know where exactly to adjust it, from what I understand it is not legal to change [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/metrics/metrics.go#L149).

Any suggestions on where it is better to adjust it?

**Comment by sanposhiho** [comment]
@felipeagger 

> from what I understand it is not legal to change [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/metrics/metrics.go#L149).

According to @dashpole 's comment, it's legal, we can do that. (although we shouldn't frequently adjust them)

**Comment by sanposhiho** [comment]
We can compare how long `scheduler_framework_extension_point_duration_seconds` of each extension point looks like currently and adjust the buckets. PostFilter or Bind should be the slowest extension point because they make API calls inside, and I guess others are more or less similar compared to those two.

**Comment by macsko** [comment]
Please also remember that `scheduler_framework_extension_point_duration_seconds` is not the only one problematic metric. We have more, e.g. `scheduler_event_handling_duration_seconds` and `scheduler_queueing_hint_execution_duration_seconds`.

**Comment by sanposhiho** [comment]
Yeah, basically we should recheck each of them. Feel like we haven't paid much attention to the buckets when configuring new metrics.
